src/go/rpk/pkg/cli/transform/deploy.go

```diff
@@ -154,6 +155,7 @@ The --var flag can be repeated to specify multiple variables like so:
 	cmd.Flags().StringVar(&fc.functionName, "name", "", "The name of the transform")
 	cmd.Flags().Var(&fc.env, "var", "Specify an environment variable in the form of KEY=VALUE")
 	cmd.Flags().StringVar(&fc.compression, "compression", "", "Output batch compression type")
+	cmd.Flags().StringVar(&fc.from_offset, "from-offset", "", "Process an input topic partition from this offset")
```

Comment:
This needs docs in the long form help text. How does this work since offsets are per partition and deploys are for topics?

Reply:
There seems to be a string that is parsed on the backend. Generally I would recommend the parsing to be in rpk so the API is structured data only.

We should also note that this only works on the first deploy.

But also is there a use case for arbitrary start offsets? Or is it just beginning?

Reply:
>docs in the long form help text

Good catch, that's my oversight

>how does this work since offsets are per partition

What I have currently is the ability to specify a positive delta from start or a negative delta from end. I would expect the most common usage to be "+00", meaning "begin processing from the start offset of each partition", so we can always add a "start" alias or something.

>also is there a use case for arbitrary start offsets?

In practice, I don't know, though I suppose the same question applies to "start from timestamp". The motivating event for exposing this was user error leading to nominal data loss; so if the general vibe is around minor disaster recover, I think the added flexibility is probably good?

>recommend the parsing to be in rpk

fair enough. i had something more structured in a previous iteration. moved to this while working through the timequery bug and sort of liked it, but if it's particularly unsavory i can wind it back.

Reply:
put the structure back. looks similar to env vars parsing now.

Reply:
this is done